src/v/s3/client.cc

```diff
@@ -734,11 +713,21 @@ ss::future<> client_pool::stop() {
 ss::future<client_pool::client_lease> client_pool::acquire() {
     gate_guard guard(_gate);
     try {
+        // If credentials have not yet been acquired, wait for them. It is
+        // possible that credentials are not initialized right after remote
+        // starts, and we have not had a response from the credentials API yet,
+        // but we have scheduled an upload. This wait ensures that when we call
+        // the storage API we have a set of valid credentials.
+        if (unlikely(!_apply_credentials)) {
+            co_await wait_for_credentials();
```

Comment:
yeah I see this is an issue. the call to `load...` comes from `remote` which is a much later commit. I cannot move that commit back because it depends on all the intermediate commits.

maybe s3 client and remote should be changed together in the same later commit.

Reply:
I think it is ok to leave it. But it's something to keep in mind for future PRs.